### ISSUE TITLE
Remove obsolete member in src/Primitives/src/InplaceStringBuilder.cs

### ISSUE THREAD

**Issue by muratg** [dotnet/runtime#35994]
Remove this obsolete member in [src/Primitives/src/InplaceStringBuilder.cs](https://github.com/aspnet/Extensions/blob/master/src/Primitives/src/InplaceStringBuilder.cs#L11):

```c#
    [Obsolete("This type is obsolete and will be removed in a future version.")]
    public struct InplaceStringBuilder
    {
```

And file a new issue in https://github.com/aspnet/Announcements/issues/ if this is a notable breaking change.

**Comment by ericstj** [comment]
@davidfowl @Pilchie what do you think about removing this?  Is that something we want to do now, or would it create an adoption barrier for 5.0 packages?  Typically we don't remove types *ever* in dotnet/runtime, so trying to see if there needs to be a different precedent for Microsoft.Extensions.

**Comment by Pilchie** [comment]
We removed several obsolete things in 3.0 that were obsoleted in 2.x.
 
A quick search through GitHub shows the majority of the instances are full copies of ASP.NET source trees.

Also, looking at NuGet, it doesn't appear that any libraries use it, so I think it's reasonable to remove.

**Comment by davidfowl** [comment]
Yes I think we should remove it.

**Comment by maryamariyan** [comment]
closed via #40743.
